Firewall/Provider.php

```diff
@@ -165,7 +166,12 @@ protected function normalizeConfig(array $config)
     public function getFirewall($configName = null, array $options = array(), Request $request = null)
     {
         if (!$request) {
-            $request = $this->container->get('request');
+            try {
+                $requestStack = $this->container->get('request_stack');
```

Comment:
Maybe you should use [`ContainerInferface::has()`](https://github.com/symfony/symfony/blob/3.0/src/Symfony/Component/DependencyInjection/ContainerInterface.php#L60) method instead of catching `ServiceNotFoundException` exception?